src/dune_config_file/feature_flags.ml

```diff
@@ -0,0 +1 @@
+let cache_enabled_by_default = true
```

Comment:
For `main` this should be `false`.

Reply:
I thought the point was to enable by default on `main`?
Or should I open a separate PR targeting the toolchains branch setting this up to true?

Reply:
I don't think we can just enable the dune cache by default on `main`, because in such case there would be no point to have a feature flag.

Reply:
I updated the feature flag. At this point I don't see how we can satisfy all these criteria:
- all unstable features are guarded by feature flags
- these features have tests that actually test the feature
- when we want to enable this feature on main, the commit to do so will only need a `- let x = false` `+ let x = true`
Maybe there's something I missed...
This PR should be mergeable in the meantime__

Reply:
Update: I removed the feature flag as this is something that we want to merge on main